src/utils/deployRollup.ts

```diff
@@ -50,6 +52,31 @@ export async function deployRollup({
     const batchPosterAddress = batchPoster.address;
     const nativeToken = rollupConfig.nativeToken;
 
+    // custom gas token
+    if (nativeToken !== zeroAddress) {
```

Comment:
New release of Orbit SDK deploys deterministic factories by default, so for custom gas token rollup, we have to approve the token to be spent by the rollup creator and deploy the factories.